### PR TITLE
ingest: Reuse Stellar-Core's cached buckets if possible.

### LINKED ISSUE #3631: Reuse Captive-Core storage path
Stellar-Core 17.1.0 allows reusing buckets for faster catchup (https://github.com/stellar/stellar-core/pull/2994). Right now we generate random subfolder at Captive-Core startup that prevents from using this feature. This should be done after protocol upgrade to ensure version 17.1.0 is installed.

### PR DESCRIPTION
**Pull request by Shaptic**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [x] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [x] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What
Stellar-Core can intelligently reuse existing bucket files (stellar/stellar-core#2994) as of v17.1.0 (see the [release notes](https://github.com/stellar/stellar-core/releases/tag/v17.1.0)). We store these under the `--captive-core-storage-path=./` directory. It used to be a randomly-generated subfolder (e.g. `captive-core-d34db33f/`) on startup; now it's kept consistent (just `captive-core/`) to take advantage of this feature.

### Why
Performance: faster startup. Closes #3631.

### Known limitations
This feature is only available with Stellar-Core >= v17.1.0. We can introduce version checks to keep having random buckets with < 17.1, but this will complicate startup significantly, since we'd now need to run & parse `stellar-core version`.

Needs #3683 to be merged first for proper 17.1 support in our builds.

### CODE REVIEW

**Comment by ire-and-curses** [comment]
> The next release of Horizon will require Stellar-Core >= v17.1.0.

What's the reason Horizon is not backwards-compatible?

**Comment by Shaptic** [comment]
@ire-and-curses I think I misspoke. More accurately: to take advantage of the performance improvement, we need >= 17.1. Otherwise, I think Core will just flush the buckets folder. I'm building an older binary as we speak to make sure it's still backwards compatible. The biggest change in both cases, though, is that the directory isn't cleaned up on shutdown anymore.

**Comment by ire-and-curses** [comment]
> Otherwise, I think Core will just flush the buckets folder.

Ok, that's what I was expecting, thanks. Yeah as you say let's confirm for sure it will still work with an older version (and just won't be any faster).

**Comment by bartekn** [comment]
> Otherwise, I think Core will just flush the buckets folder. I'm building an older binary as we speak to make sure it's still backwards compatible.

Have you checked this? I think Stellar-Core does not flush the folder between runs.

I checked the [failing test](https://app.circleci.com/pipelines/github/stellar/go/8432/workflows/a2b2d1d7-2968-4c3b-873d-8607d1b65eac/jobs/46170): `TestReingestDB`. I think the problem is that the test starts Horizon normally in the background somewhere (maybe by our integration test framework) because the log output doesn't make sense. In reingestion we don't ingest state nor verify state but the logs suggest something else. I think if we disable starting Horizon before we actually run reingestion command it should work. It's possible that it worked before because different (random) bucket folder was used.

**Comment by Shaptic** [comment]
> I think Stellar-Core does not flush the folder between runs.

I tested with 17.0 and confirmed that it does delete and redownload the bucket files @bartekn.

> I think the problem is that the test starts Horizon normally in the background somewhere

Yep, I came to the same exact conclusion and introduced `IntegrationTest.StopHorizon()` to alleviate this in f9aaba985a033bdf5b657cabe89ef139ea0f3770. All tests pass now!

File: ingest/CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,9 @@ All notable changes to this project will be documented in this file. This projec
 
 ## Unreleased
 
+### New Features
+* **Performance improvement**: the Captive Core backend now reuses bucket files whenever it finds existing ones in the corresponding `--captive-core-storage-path` (introduced in [v2.0](#v2.0.0)) rather than generating a one-time temporary sub-directory ([#3670](https://github.com/stellar/go/pull/3670)). Note that taking advantage of this feature requires [Stellar-Core v17.1.0](https://github.com/stellar/stellar-core/releases/tag/v17.1.0) or later.
```
Comment (bartekn) [inline ingest/CHANGELOG.md:9]: Shouldn't we use value from `Config` instead of `--captive-core-storage-path`?

**Review by bartekn — APPROVED** [review]
> I tested with 17.0 and confirmed that it does delete and redownload the bucket files @bartekn.

Sounds good! On the contrary 17.1 does not delete the buckets, correct?

LGTM!

**Comment by Shaptic** [comment]
@bartekn correct! The catchup is much faster as expected.

**Pull request merged**